### ISSUE TITLE
Swagger based on v2 WP-API

### ISSUE THREAD

**Issue by starfishmod** [WP-API/WP-API#2480]
I have built a Swagger file that documents the current API as described at http://v2.wp-api.org/

you can find this at https://swaggerhub.com/api/starfishmod/Wordpress/0.1.0

I mention this for the following reasons:
- perhaps other will find this useful
- how much a leap would it be to include this in the plugin and auto-generate the few missing parts (host and schemes) so people could call http://wpexample.com/wp/v2/swagger to return the swagger file to pass onto others?
- if there are any errors or omissions please let me know

**Comment by danielbachhuber** [comment]
> I have built a Swagger file that documents the current API

Cool! Did you do it manually, or write a script for it? If the latter, are you open to sharing?

> how much a leap would it be to include this in the plugin and auto-generate the few missing parts (host and schemes) so people could call http://wpexample.com/wp/v2/swagger to return the swagger file to pass onto others?

While I don't think it quite makes sense at this point to bundle in the main plugin, I think it would be a really cool [secondary plugin](http://v2.wp-api.org/guide/plugins/).

**Comment by starfishmod** [comment]
Honestly I hand coded - however now that I have done it that way I think I would prefer to generate it based on the endpoint information. Also from further reviewing the code I think it would be better as a separate plugin too.

If I have time I will code something up in the next few days.

**Comment by starfishmod** [comment]
Ok so here is the auto generate swagger from restserver working as a plugin
https://github.com/starfishmod/WPAPI-SwaggerGenerator
I would appreciate any feedback etc :)

**Comment by kadamwhite** [comment]
@BE-Webdesign why'd this get closed out? I'm still interested in having the discussion around providing a swagger specification, it could be beneficial to be a part of this: https://openapis.org/specification

**Comment by starfishmod** [comment]
@kadamwhite I agree it would be nice to have this as part of the plugin - but there are some issues that I had to work around which are a disconnect between my plugin and this one.
The biggest being that OpenAPI wants only Oauth 2.0/ApiKey/Basic HTTP auth schemes while the WP-APIv2 only really supports Oauth 1.0 and Header nounce(?? going off memory here) auth - which is not really useful for B2B stuff.
The are other issues such as HTTP return codes as well that I have to "hack" around that would require changes in the code here.

**Comment by BE-Webdesign** [comment]
In a slack conversation today, it was deemed this issue would need to wait until other priorities were addressed.  A Swagger implementation can still exist as a plugin.  In the future, this should definitely be revisited.
